### DISCUSSION TITLE
CUTLASS based FMHA kernels

### DISCUSSION THREAD

**Discussion by hwu36** [NVIDIA/cutlass#1003] [Show and tell]
First, @danthe3rd synced xFormers FMHA to CUTLASS repositories earlier last week.  The new version is up to 20% faster in the forward pass and up to 10x faster in the backward pass.  See #992 for the detail.

Second, @tridao just released FlashAttention v2.  Compared with v1, v2 is 2x faster.  See their [repository](https://github.com/Dao-AILab/flash-attention) for the detail.

Both implementations can run on both A100 and H100.

**Comment by MARD1NO** [comment]
I wonder if cutlass will develop "official" FMHA Kernel?

**Reply by hwu36** [reply]
User xformers or flashv2.  They are awesome.  We helped them on their implementations.  No need for us to do it again.
